src/assets/scripts/client/aircraft/Pilot/Pilot.js

```diff
@@ -488,12 +488,19 @@ export default class Pilot {
      *
      * @for Pilot
      * @method descendViaStar
-     * @param altitude {number}  (optional) altitude at which the descent will end (regardless of fix restrictions)
-     *                                      this should be the altitude of the lowest fix restriction on the STAR
-     * @return {array}           [success of operation, readback]
+     * @param bottomAltitude {number} (optional) altitude at which the descent will end (regardless of fix restrictions)
+     * @return {array}                [success of operation, readback]
      */
-    descendViaStar(altitude = 0) {
-        this._mcp.setAltitudeFieldValue(altitude);
+    descendViaStar(bottomAltitude) {
+        if (bottomAltitude === undefined) {
+            bottomAltitude = this._fms.getBottomAltitude();
+        }
+
+        if (_isNil(bottomAltitude) || bottomAltitude === Infinity) {
```

Comment:
Isn't the `_isNil()` duplicative here?

Reply:
that was an attempt to be defensive in case `.getBottomAltitude()` returned a non-numeric value, which it can in many cases.